lib/pbench/server/api/resources/auth.py

```diff
@@ -0,0 +1,94 @@
+import jwt
+import os
+import datetime
+from flask import session, request, abort
+from flask_httpauth import HTTPTokenAuth
+from pbench.server.api.resources.models import UserModel
+
+
+auth = HTTPTokenAuth("Bearer")
+
+
+def encode_auth_token(token_expire_duration, user_id, logger):
+    """
+    Generates the Auth Token
+    :return: string
+    """
+    payload = {
+        "iat": datetime.datetime.utcnow(),
+        "exp": datetime.datetime.utcnow()
+        + datetime.timedelta(minutes=int(token_expire_duration)),
+        "sub": user_id,
+    }
+    try:
+        # Get jwt key
+        jwt_key = get_secret_key()
+        return jwt.encode(payload, jwt_key, algorithm="HS256")
+    except Exception as e:
+        logger.info(
+            f"Exception occurred while encoding the auth token for user_id {user_id}, exception: {e}"
+        )
+        return None
+
+
+@auth.verify_token
+def verify_token(auth_token):
+    """
+    Validates the auth token
+    :param auth_token:
+    :param app:
+    :return: integer|string
+    """
+    try:
+        payload = jwt.decode(auth_token, get_secret_key(), algorithms="HS256")
+        user_id = payload["sub"]
+        if user_id not in session:
+            return False
+        if auth_token not in session[user_id]:
+            return False
+        user = UserModel.query.filter_by(id=user_id).first()
+        return user
+    except jwt.ExpiredSignatureError:
+        print("Signature expired")
+        return False
+    except jwt.InvalidTokenError:
+        print("Invalid token")
+        return False
+    except Exception as e:
+        return False
+
+
+def get_secret_key():
+    try:
+        return os.getenv("SECRET_KEY", "my_precious")
+    except Exception as e:
+        print(f"{__name__}: ERROR: {e.__traceback__}")
+
+
+def get_auth_token(logger):
+    # get auth token
+    auth_header = request.headers.get("Authorization")
+
+    if not auth_header:
+        logger.error(f"Auth token not provided during logout")
+        abort(
+            403,
+            message="Please add 'Authorization' token as Authorization: Bearer <JWT_Auth_token>",
+        )
+
+    try:
+        auth_type, auth_token = auth_header.split()
```

Comment:
We dont really need a custom `get_auth_token` method since we are using an `@auth.login_required` decorator on each method where we wanted to check the auth token. We did this to protect the endpoint but we are now any using the `HTTPTokenAuth("Bearer")` auth instance so we dont need to re-validate using `get_auth_token`. However, I kept it here for now just in case